src/snapred/backend/recipe/algorithm/MakeDirtyDish.py

```diff
@@ -18,10 +18,19 @@ def PyInit(self):
         self.declareProperty("InputWorkspace", defaultValue="", direction=Direction.Input)  # noqa: F821
         self.declareProperty("OutputWorkspace", defaultValue="", direction=Direction.Output)  # noqa: F821
         self.setRethrows(True)
-        self._CISmode: bool = Config["cis_mode"]
+
+        cisModeConfig = Config["cis_mode"]
+        self.enabled: bool = cisModeConfig.get("enabled")
+        self.preserve: bool = cisModeConfig.get("preserveDiagnosticWorkspaces")
+
+        if self.enabled != self.preserve:
```

Comment:
Preserve is allowed to be off while enabled is on.
My suggestion for testing this was to confirm that the diagnostic workspaces were not preserved if either "cis_mode" was disabled or the sub-flag of "preserveDiagnosticWorkspaces" was disabled.

Please remove these warnings.